docs/source/java/examples/quickstartguide.rst

```diff
@@ -0,0 +1,277 @@
+        varCharVector.allocateNew(3);
+        varCharVector.set(0, "one".getBytes());
+        varCharVector.set(1, "two".getBytes());
+        varCharVector.set(2, "three".getBytes());
+        varCharVector.setValueCount(3);
+        System.out.println("Vector created in memory: " + varCharVector);
+    }
+
+Create A Field
+**************
+
+Fields are used to denote the particular columns of tabular data.
+
+**Field**: Create a column "document" of string type with metadata.
+
+.. code-block:: Java
+
+    import org.apache.arrow.vector.types.pojo.ArrowType;
+    import org.apache.arrow.vector.types.pojo.Field;
+    import org.apache.arrow.vector.types.pojo.FieldType;
+
+    Map<String, String> metadata = new HashMap<>();
+    metadata.put("A", "Id card");
+    metadata.put("B", "Passport");
+    metadata.put("C", "Visa");
+    Field document = new Field("document", new FieldType(true, new ArrowType.Utf8(), /*dictionary*/ null, metadata), /*children*/ null);
+
+Create A Schema
+***************
+
+Schema holds a sequence of fields together with some optional metadata.
+
+**Schema**: Create a schema describing datasets with two columns:
+a int32 column "A" and a utf8-encoded string column "B"
+
+.. code-block:: Java
+
+    import org.apache.arrow.vector.types.pojo.ArrowType;
+    import org.apache.arrow.vector.types.pojo.Field;
+    import org.apache.arrow.vector.types.pojo.FieldType;
+    import org.apache.arrow.vector.types.pojo.Schema;
+    import static java.util.Arrays.asList;
+
+    Map<String, String> metadata = new HashMap<>();
+    metadata.put("K1", "V1");
+    metadata.put("K2", "V2");
+    Field a = new Field("A", FieldType.nullable(new ArrowType.Int(32, true)), null);
+    Field b = new Field("B", FieldType.nullable(new ArrowType.Utf8()), null);
+    Schema schema = new Schema(asList(a, b), metadata);
+
+Create A VectorSchemaRoot
+***************************
+
+VectorSchemaRoot is somewhat analogous to tables and record batches in the other
+Arrow implementations.
+
+**VectorSchemaRoot**: Create a dataset with metadata that contains integer age and
+string names of data.
+
+.. code-block:: Java
+
+    import org.apache.arrow.memory.BufferAllocator;
+    import org.apache.arrow.memory.RootAllocator;
+    import org.apache.arrow.vector.IntVector;
+    import org.apache.arrow.vector.VarCharVector;
+    import org.apache.arrow.vector.VectorSchemaRoot;
+    import org.apache.arrow.vector.types.pojo.ArrowType;
+    import org.apache.arrow.vector.types.pojo.Field;
+    import org.apache.arrow.vector.types.pojo.FieldType;
+    import org.apache.arrow.vector.types.pojo.Schema;
+
+    import java.nio.charset.StandardCharsets;
+    import java.util.HashMap;
+    import java.util.Map;
+    import static java.util.Arrays.asList;
+
+    Map<String, String> metadataField = new HashMap<>();
+    metadataField.put("K1-Field", "K1F1");
+    metadataField.put("K2-Field", "K2F2");
```

Comment:
IMO let's not put metadata in the example here, let's try to keep the example minimal.

Reply:
Deleted